File: FinBoard/BackEnd/src/main/java/pwork/greco/antonio/finboard/entity/Order.java
package pwork.greco.antonio.finboard.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "[Order]")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "PortfolioId", nullable = false)
    private Portfolio portfolio;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "InstrumentId", nullable = false)
    private Instrument instrument;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "OperatorId", nullable = false)
    private User operator;

    @Column(nullable = false, length = 4)
    private String side; // buy/sell

    @Column(nullable = false, precision = 18, scale = 4)
    private BigDecimal quantity;

    @Column(precision = 18, scale = 4)
    private BigDecimal price;

    @Column(nullable = false, length = 20)
    private String status;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    private LocalDateTime executedAt;

    @Column(nullable = false)
    private Boolean deleted;
}

File: FinBoard/FrontEnd/src/app/models/Order.ts
import { Instrument } from './Instrument';
import { Portfolio } from './Portfolio';
import { User } from './User';
//
//export interface Order {
//  id: number;                    // bigint
//  createdAt: string;             // datetime2(6)
//  deleted: boolean;              // bit
//  executedAt?: string | null;    // datetime2(6) nullable
//  price?: number | null;         // numeric(18,4) nullable
//  quantity: number;              // numeric(18,4)
//  side: string;                  // varchar(4)
//  status: string;                // varchar(20)
//  instrumentId: number;          // bigint (FK)
//  operatorId: number;            // bigint (FK)
//  portfolioId: number;           // bigint (FK)
//}

export interface Order {
  id: number;
  createdAt: string;
  deleted: boolean;
  executedAt?: string | null;
  price?: number | null;
  quantity: number;
  side: string;
  status: string;
  instrument: Instrument;
  user: User;
  portfolio: Portfolio;
  operationDate: Date;
  evaluationDate: Date;
}